pkg/restore/restore.go

```diff
@@ -634,20 +647,33 @@ func (ctx *context) restoreResource(resource, namespace, resourcePath string) (a
 		}
 
 		// necessary because we may have remapped the namespace
-		obj.SetNamespace(namespace)
+		// if the namespace is blank, don't create the key
+		if namespace != "" {
+			obj.SetNamespace(namespace)
+		}
 
 		// add an ark-restore label to each resource for easy ID
 		addLabel(obj, api.RestoreLabelKey, ctx.restore.Name)
 
 		ctx.infof("Restoring %s: %v", obj.GroupVersionKind().Kind, obj.GetName())
-		_, err = resourceClient.Create(obj)
-		if apierrors.IsAlreadyExists(err) {
-			addToResult(&warnings, namespace, err)
+		_, restoreErr := resourceClient.Create(obj)
+		if apierrors.IsAlreadyExists(restoreErr) {
+			equal, err := ctx.objectsAreEqual(obj, resourceClient, restoreErr)
+			// Log any errors trying to resolve the conflict; nil means no conflic
+			if err != nil {
+				addToResult(&warnings, namespace, err)
+			}
+			if equal {
+				ctx.infof("Backup for %s: %v matches cluster version, skipping restore.", obj.GroupVersionKind(), obj.GetName())
+			} else {
+				e := fmt.Errorf("%s and backup does not match.", restoreErr)
+				addToResult(&warnings, namespace, e)
+			}
 			continue
 		}
-		if err != nil {
+		if restoreErr != nil {
```

Comment:
Please add a comment indicating that if we get here, we got some error other than already-exists